upf/upf_forward.c

```diff
@@ -211,10 +212,15 @@ upf_forward (vlib_main_t * vm, vlib_node_runtime_t * node,
 		  fp_entry =
 		    pool_elt_at_index (gtm->upf_forwarding_policies,
 				       far->forward.fp_pool_index);
-		  vnet_buffer (b)->ip.adj_index[VLIB_TX] =
-		    fp_entry->dpo.dpoi_index;
+
+		  vec_foreach (rpath, fp_entry->rpaths)
+		  {
+		    if (rpath->frp_proto ==
+			(is_ip4 ? DPO_PROTO_IP4 : DPO_PROTO_IP6))
+		      break;
+		  }
```

Comment:
what happens when there is entry that matches the current protocol version?

If nothing matches, then rpath will end up pointing to a element after the end of vector. That is an invalid entry and the index value in there is invalid.

Reply:
Very good point, thanks.
If policy is underconfigured it will lead to crash here.
Fixed this to use normal forwarding based on FAR's NWI.